src/shared/sym_load.c

```diff
@@ -46,6 +46,22 @@ void* so_get_module_handle(const char *so){
 #endif
 }
 
+void* so_check_module_loaded(const char *so){
+#ifdef WIN32
+    char file_name[MAX_PATH] = { 0 };
+    snprintf(file_name, MAX_PATH-1, "%s.dll", so);
+    return GetModuleHandle(file_name);
```

Comment:
What about this? 
https://learn.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-getmodulehandlea

> The GetModuleHandle function returns a handle to a mapped module without incrementing its reference count. However, if this handle is passed to the [FreeLibrary](https://learn.microsoft.com/en-us/windows/desktop/api/libloaderapi/nf-libloaderapi-freelibrary) function, the reference count of the mapped module will be decremented. Therefore, do not pass a handle returned by GetModuleHandle to the FreeLibrary function. Doing so can cause a DLL module to be unmapped prematurely.

Reply:
I added a comment in the header, no one should call `FreeLibrary` with this handle
https://github.com/wazuh/wazuh/blob/f265a7a969ab4eaa1422e71620669f605df57a8a/src/headers/sym_load.h#L23-L29

Do you think we should be more explicit?